test/jdk/java/lang/invoke/MethodTypeTest.java

```diff
@@ -229,6 +229,7 @@ public String[] badMethodDescriptorStrings() {
                 "([V)V",
                 "(" + "[".repeat(256) + "J)I",
                 "(java/lang/Object)V",
+                "()java/lang/Object",
         };
```

Comment:
And maybe also add a test case for an array type with too many dimensions in the return type:
```suggestion
                "()java/lang/Object",
                "()" + "[".repeat(256) + "Ljava/lang/Object;",
        };
```